### ISSUE TITLE
Length properties export with imperial conversion wrongly applied

### ISSUE THREAD

**Issue by Moult** [Autodesk/revit-ifc#478]
Potentially the same as #199, but maybe not.

1. Add new instance parameter "Foo" of type Number
2. Fill out number with value, say 4000
3. Export IFC using user defined parameter sets config of "Foo Length"
4. IFC exports the parameter of Foo with a value of 1219200 instead of 4000

Suspiciously 1219200 / 4000 = 304.8, so it looks like a unit conversion.

Note this doesn't happen when the parameter is of type Length.

**Comment by AngelVelezSosa** [comment]
So what should happen in this case?  Should the value be taken at face value, so that if you export with display type meters vs. exporting with display type millimeters, it stays as "4000"?  Is this user error in the sense that there isn't enough information to make an informed decision here?

**Comment by Moult** [comment]
Yes taking a "number" as face value sounds like a sensible solution, given that the user has not provided units.

**Comment by ArtemHnatko** [comment]
The issue could not be reproduced in Revit 2023–2026. The custom number parameter exports correctly without incorrect unit conversion.

**Testing environment:**
- Autodesk Revit 2023 (IFC 23.4.1.0)
- Autodesk Revit 2024 (IFC 24.3.20.0)
- Autodesk Revit 2025 (IFC 25.4.0.27)
- Autodesk Revit 2026 (IFC 26.1.0.23)

![Image](https://github.com/user-attachments/assets/3b104e71-49b8-4c49-9bb8-76384754e6a9)
